code/script.js

```diff
@@ -1,51 +1,452 @@
 // Declare your functions after this comment
 
+
+//Iteration 1
+//part 1
 // This function will add a chat bubble in the correct place based on who the sender is
 const showMessage = (message, sender) => {
-  // the if statement checks if the sender is 'user' and if that's the case it inserts an html senction inside the chat with the posted message
+  // the if statement checks if the sender is 'user' and if that's the case it inserts an html section inside the chat with the posted message
   if (sender === 'user') {
     chat.innerHTML += `
-      <section class="user-msg">
-        <div class="bubble user-bubble">
-          <p>${message}</p>
-        </div>
-        <img src="assets/user.png" alt="User" />  
-      </section>
+        <section class="user-msg" id="user-msg">
+          <div class="bubble user-bubble">
+            <p>${message}</p>
+          </div>
+          <img src="./assets/user.png" alt="user bot">
+        </section>
     `
+
     // the else if statement checks if the sender is a bot and if that's the case it inserts an html senction inside the chat with the posted message
   } else if (sender === 'bot') {
     chat.innerHTML += `
-      <section class="bot-msg">
-        <img src="assets/bot.png" alt="Bot" />
-        <div class="bubble bot-bubble">
-          <p>${message}</p>
-        </div>
-      </section>
-    `
+        <section class="bot-msg" id="bot-msg">
+          <img src="./assets/bot.png" alt="chat bot">
+          <div class="bubble bot-bubble">
+            <p>${message}</p>
+          </div>
+        </section>
+        `
   }
-  // This little thing makes the chat scroll to the last message when there are too many to be shown in the chat box
   chat.scrollTop = chat.scrollHeight
 }
 
-// Starts here
+//part 2
+// When website is loaded, chatbot asks first question.
 const greetUser = () => {
   // here we call the function showMessage, that we declared earlier with the argument "Hello there, What's your name?" for message, and the argument "bot" for sender
-  showMessage("Hello there, What's your name?", 'bot')
-  // Just to check it out, change 'bot' to 'user' here 👆
+  showMessage("Greetings dearest customer.  To proceed, please enter your name.", 'bot')
+}
+// Just to check it out, change 'bot' to 'user' here
+
+//Iteration 2
+//username pops up on chat on user's side
+const showName = (event) => {
+  event.preventDefault();
+  // when you submit a form, the default action is to send the form data to the server and reload the page.  By calling event.preventDefault() within an event handler, you can stop the browser from performing its default action and allow you to handle the event in a custom manner.
+  userName = nameInput.value;
+  showMessage(`${userName}`, "user");
+  nameInput.value = ""; //clear nameInput's value 
+  setTimeout(proposeActivity, 1000);
+}
+
+
+//Iteration 3
+//part 1
+//bot replies back using message proposing activities
+const proposeActivity = () => {
```

Comment:
I like the way you have paced the pop up of messages and the buttons/drop-down menus overall. i think that creates a nice user experience. For the name input, I can add my name a second time before the buttons appear since it stays a bit longer but I'm not sure if removing them earlier would be better either.